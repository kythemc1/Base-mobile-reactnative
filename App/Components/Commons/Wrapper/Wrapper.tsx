import {View, Text} from 'react-native-ui-lib';
import React from 'react';
import {
    KeyboardAvoidingView,
    Platform,
    StatusBar,
    StyleSheet,
} from 'react-native';
import {getStatusBarHeight} from 'react-native-status-bar-height';

export default function Wrapper({
    children,
    avoidStatusBar = false,
    disableKeyboardAvoidingView = false,
    backgroundColor = 'transparent',
}: {
    children: React.ReactNode;
    avoidStatusBar?: boolean;
    disableKeyboardAvoidingView?: boolean;
    backgroundColor?: string;
}) {
    return (
        <View flex backgroundColor={backgroundColor}>
            {avoidStatusBar && (
                <StatusBar barStyle="light-content" translucent animated/>
            )}
            {!disableKeyboardAvoidingView && (
                <KeyboardAvoidingView
                    behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
                    style={StyleSheet.absoluteFill}>
                    {children}
                </KeyboardAvoidingView>
            )}
            {/* <KeyboardAvoidingView
        style={style.container}
        behavior="padding"
        enabled={Platform.OS === 'ios' && !disableKeyboardAvoidingView}>
        {!!avoidStatusBar && <View style={style.avoidStatusBar} />}
        <View flex>{children}</View>
      </KeyboardAvoidingView> */}
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#FFFFFF',
    },
    avoidStatusBar: {
        height: getStatusBarHeight(),
    },
});
