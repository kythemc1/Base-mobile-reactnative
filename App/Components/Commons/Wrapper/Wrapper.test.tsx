import renderer from 'react-test-renderer';
import React from 'react';
import Wrapper from './Wrapper';
test('renders correctly', () => {
    const tree = renderer.create(<Wrapper children={undefined} />).toJSON();
    expect(tree).toMatchSnapshot();
});
