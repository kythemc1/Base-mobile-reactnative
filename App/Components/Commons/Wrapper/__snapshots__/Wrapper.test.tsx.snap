// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
      },
      undefined,
      {
        "flex": 1,
      },
      undefined,
      {},
      {},
      {},
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  />
</View>
`;
